tests/test_convolution_1d.py

```diff
@@ -0,0 +1,194 @@
+import pytest
+import numpy as np
+
+import neurostatslib.utils as utils
+
+
+class Test1DConvolution:
+    @pytest.mark.parametrize("basis_matrix", [np.zeros((n, n)) for n in [0, 1, 2]])
+    @pytest.mark.parametrize("trial_count_shape", [(1, 30, 2), (2, 10, 20)])
+    def test_basis_matrix_type(self, basis_matrix, trial_count_shape: tuple[int]):
+        vec = np.ones(trial_count_shape)
+        raise_exception = any(k == 0 for k in basis_matrix.shape)
+        if raise_exception:
+            with pytest.raises(ValueError, match=r"Empty basis_matrix provided\. "
```

Comment:
doesn't `r` make it a raw string? why use that here?

Reply:
this is a leftover after the line was split in two. it is needed for the next line but it can be removed